### PR TITLE
[READY] Janitor balance and QoL changes

### PR DESCRIPTION
**Pull request by cacogen**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: cacogen
balance: Space cleaner in spray bottles travels 5 tiles instead of 3 (doesn't affect other reagents)
balance: Trash bag fits in exosuit slot of janitor biosuit
balance: All soaps are faster
balance: Soaps now clean all decals on a tile instead of just one
balance: Soap now has limited uses (100 for most, 300 for NT brand which janitor gets)
balance: Mops are considerably faster
balance: Basic mop holds twice the reagents  
balance: Janitorial cart now refills mops completely with one click
balance: Basic and advanced mops are more robust (8 force and 12 force)
balance: Galoshes no longer leave bloody footprints
balance: Chameleon noslips no longer leave bloody footprints
fix: Bulb boxes can be used on light replacers refill them in bulk
balance: Light replacer can be used in-hand to change all bulbs on a tile
add:  Custodial barrier projector which creates solid wet floor signs that force people to walk to pass (available through service protolathe)
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

Janitor's been an exercise in futility for years, and the job is made redundant by janitor cyborgs who can't even keep up with the mess themselves. Spreading blood is instant, but takes a lot longer to clean. The goal here is to make it easier to keep up with the mess and make the controls less finicky while avoiding affecting combat too much (the ~~mop and~~ spray bottle ~~are~~ is a good weapon~~s~~ after all).

To-do:
- [x] make soap buff less extreme
- [x] remove use of integrity var
- [x] rename density vars to be more specific

What else I'd like to add:
- [x] make mops faster
- [x] increase reagents basic mop can hold
- [x] make holosigns force people to walk to pass
- [ ] ~~make mops/cleaner cover more tiles at once?~~

Here is an only slightly autistic and nearly exhaustive table of changes:

| Object                            | Change                          | Old      | New          |
| --------------------------------- | ------------------------------- | -------- | ------------ |
| **Mop**                           | Clean speed                     | 3 sec    | 1.5 sec      |
|                                   | Reagents held                   | 5u       | 15u          |
|                                   | Force                           | 3        | 8            |
|                                   | Throw force                     | 5        | 10           |
| **Advanced mop**                  | Clean speed                     | 2 sec    | 0.8 sec      |
|                                   | Force                           | 6        | 12           |
|                                   | Throw force                     | 8        | 14           |
| **Soap**                          | Clean speed                     | 5 secs   | 3.5 secs     |
|                                   | Uses                            | ∞        | 100          |
|                                   | In janidrobe                    | Yes      | No           |
|                                   | Decals cleaned per use          | 1        | All on tile  |
| **NT soap**                       | Clean speed                     | 5 secs   | 2.8 secs     |
|                                   | Uses                            | ∞        | 300          |
|                                   | In janidrobe                    | No       | Yes          |
|                                   | Decals cleaned per use          | 1        | All on tile  |
| **Homemade soap**                 | Clean speed                     | 4.5 secs | 3 secs       |
|                                   | Uses                            | ∞        | 100          |
|                                   | Decals cleaned per use          | 1        | All on tile  |
| **Deluxe soap**                   | Clean speed                     | 4 secs   | 2 secs       |
|                                   | Uses                            | ∞        | 100          |
|                                   | Decals cleaned per use          | 1        | All on tile  |
| **Syndie soap**                   | Clean speed                     | 1 sec    | 0.5 sec      |
|                                   | Uses                            | ∞        | 100          |
|                                   | Decals cleaned per use          | 1        | All on tile  |
| **Light replacer**                | Use in-hand instead of clicking | No       | Yes          |
|                                   | Use bulb box on to refill       | No       | Yes          |
| **Wet floor barrier projector**   | Exists                          | No       | yuh          |
| **Janitorial cart**               | Units added to mop per click    | 5u       | Mop capacity |
| **Galoshes**                      | Can leave bloody footprints     | Yes      | No           |
| **Chameleon noslips**             | Can leave bloody footprints     | Yes      | No           |
| **Janitor biosuit**               | Trash bag fits in exosuit storage    | No       | Yes          |
| **Space cleaner in spray bottle** | Distance                        | 3 tiles  | 5 tiles      |

### CODE REVIEW

**Comment by XDTM** [comment]
This seems to make the mop even less useful in comparison with the other cleaning methods. Since it's pretty much a signature tool for janitors it would be a pity to discourage using it.

**Comment by ExcessiveUseOfCobblestone** [comment]
The mop is a very cruddy version of any reagent container, unless it has some force meme like the birthday hat i'm not sure how it's a good weapon.

Remove all cleaning tools but mop and parody the mining job by having upgrades on the ~~mop~~kinetic cleaner.

**Comment by XDTM** [comment]
Mop trophies from cleaning really weird stuff

**Comment by subject217** [comment]
Not a fan of the soap change. Maybe if you buff the mop without making it a super slip machine.

**Comment by ExcessiveUseOfCobblestone** [comment]
Megamesses with special cleaning requirements???

**Comment by Mickyan** [comment]
I don't mind this (I mean, some of the changes seem a bit bizarre but I haven't tested it to see how it works in practice) but the root of the problem is the way blood decals are created is awful and overbuffing some of the janitor tools seems like a roundabout way to deal with the issue

Ex: if you get punched a couple of times and start bleeding, stand still for five seconds you create a blood decal that can fill an entire room with footprints and there's no difference between that and leaving a body to bleed out for an hour

**Comment by Time-Green** [comment]
Make mops clean in a 3x3 radius if you're looking to make it more interesting. Most people already click every tile in a 3x3 radius. It would make soap great for small stuff, and mops for bigger messes

**Comment by Teepeepee** [comment]
cleaning all decals on a tile at once is the only good thing about the mop, don't ruin it like that, give mop some love too!

**Comment by Kagus** [comment]
Mop definitely needs some love, especially seeing as the most effective way of using it (spam-clicking the 3x3 area around you to queue up all the tiles) will run out of water halfway through your second round and you need to re-tag half the tiles if you're not refilling after every grid. Oh, also, remember that slipping is getting kinda-nerfed in the upcoming combat rework.

3x3 on a single use maybe is a bit much, but I could definitely see something like a directional 1x3 such as the mech drill mining.

Turning janitor into an on-station miner, fighting an invisible war against messes would... Definitely be a thing. Give them a role-related brain trauma that makes them see demonic dust bunnies everywhere or something. Would kinda make it a lot scarier of a role to shove the newbies into though, and it's really a pretty good job for someone new to the game. Not sure if that's something we're looking to sacrifice or no.

---
So is the spray bottle toggling fixed now? Used to be that the "stream" setting would use 2u and travel 3 tiles, while the "spray" would use 5u and travel one tile.

**Comment by Therapist1** [comment]
Advanced mop charges for every mess mopped. Upon reaching 20 charges, unleash a wave of cleanliness and reset to zero.

**Comment by kriskog** [comment]
I already get beat up and mugged by captains for using the mop, imagine if soap was actually better/faster as well. 

The numbers on the spray and backpack seem absurdly high. If you want a point of reference, each chem dispenser starts out with power for 1000u of chemicals, and the starting big beakers are 100u. Buckets are 70u.
To be honest, I dont understand why the spray and backpack numers need to change at all, what does it add? It seems to be unnecessary, and creates imbalances elsewhere. Imagine a chemist getting to spray 30u of a chemical on someone in a single spray.

@Ziiro The advanced mop is already quite fast, and generates it own water. A few advanced mop in your pack, and you are a very productive janitor. Janiborg will still steal you job though.

**Comment by BeeSting12** [comment]
If space cleaner spray travels for five tiles now, I'm assuming that makes other chemicals as well? Not a good change, a buff to poison chemical sprays.

**Comment by subject217** [comment]
@BeeSting12 Naksu already nerfed spray bottles into oblivion, the range doesn't mean anything.

**Comment by Supermichael777** [comment]
Acid is already nerfed into the ground. sprays already suck

**Comment by Teepeepee** [comment]
isn't healsniping a thing with syringe guns already?

**Comment by ExcessiveUseOfCobblestone** [comment]
no because people assume it's bad and kill you before they realize they've gained health instead of lose it :^)

**Comment by Kagus** [comment]
That, and syringe guns deal small amounts of damage when hitting someone, thus making the healing element lose some effectiveness.

Plus, who the heck loads syringes with heals? You keep that gun loaded with some heavy-duty tranqs in case some greyshirt comes along and tries to eat the medbay doors or some shit. Also, the injected chems are just straight up inferior to topical medications, so why spend time and precious dispenser juice on them?

As with everything else medbay/chem-related, botany does it better with hypodermic prickle gaia. Medbay is just trying to keep its head above water, dammit.

File: code/game/atoms.dm [hunk 1]
```diff
@@ -260,17 +260,27 @@
...
+							to_chat(user, "[round(R.volume / 1000, 0.01)] kilounits of [R.name]")
 				else //Otherwise, just show the total volume
 					var/total_volume = 0
 					for(var/datum/reagent/R in reagents.reagent_list)
 						total_volume += R.volume
-					to_chat(user, "[total_volume] units of various reagents")
+					if(total_volume < 1000)
+						to_chat(user, "[total_volume] units of various reagents")
+					else
+						to_chat(user, "[round(total_volume / 1000, 0.01)] kilounits of various reagents")
```
Comment (yoyobatty) [inline code/game/atoms.dm]: replace lines 271 - 274 with:

```to_chat(user, "[total_volume%1000]?"[round(total_volume / 1000, 0.01)] kilounits":"[total_volume] units" of various reagents")```

File: code/game/atoms.dm [hunk 2]
```diff
@@ -260,17 +260,27 @@
...
 			else
 				to_chat(user, "Nothing.")
 		else if(container_type & AMOUNT_VISIBLE)
 			if(reagents.total_volume)
-				to_chat(user, "<span class='notice'>It has [reagents.total_volume] unit\s left.</span>")
+				if(reagents.total_volume < 1000)
+					to_chat(user, "<span class='notice'>It has [reagents.total_volume] unit\s left.</span>")
+				else
+					to_chat(user, "<span class='notice'>It has [round(reagents.total_volume / 1000, 0.01)] kilounits left.</span>")
+
```
Comment (yoyobatty) [inline code/game/atoms.dm]: replace lines 279 - 282 with:

to_chat(user, "[total_volume%1000]?"[round(total_volume / 1000, 0.01)] kilounits":"[total_volume] units" of various reagents")

File: code/game/atoms.dm [hunk 3]
```diff
@@ -260,17 +260,27 @@
 			if(reagents.reagent_list.len)
 				if(user.can_see_reagents()) //Show each individual reagent
 					for(var/datum/reagent/R in reagents.reagent_list)
-						to_chat(user, "[R.volume] units of [R.name]")
+						if(R.volume < 1000)
+							to_chat(user, "[R.volume] units of [R.name]")
+						else
+							to_chat(user, "[round(R.volume / 1000, 0.01)] kilounits of [R.name]")
```
Comment (yoyobatty) [inline code/game/atoms.dm]: replace lines 263 - 266 with:

to_chat(user, "[total_volume%1000]?"[round(total_volume / 1000, 0.01)] kilounits":"[total_volume] units" of various reagents")

File: code/game/atoms.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/atoms.dm]: need to use `* 0.001` per contributing.md

Comment (yoyobatty) [reply code/game/atoms.dm]: ah yes, hence, per 1000

**Comment by Vidiveni** [comment]
Space cleaner spray bottle seems to have been bugged for some time. Setting it to 5u will set range to 1 tile, and 2u to 2 tiles. Feels reversed. Don't know if there is an open issue and I'm too lazy to check.

File: code/game/atoms.dm (hunk 1 above)
Comment (octareenroon91) [reply code/game/atoms.dm]: I don't like this modification. Also are you sure about using modulo 1000?

**Comment by Potato-Masher** [comment]
Was expecting round-start Mop-Fu training, severely disappointed in that area.

Otherwise, I like the changes.

**Comment by Bawhoppen** [comment]
I can't say that I'm a necessarily a fan of making forms of work like this easier. There needs to be some level of effort required to make jobs meaningful. While this one little change won't make or break it, it's another form of powercreep, which slowly adds up.
(^Referring to the space cleaner buff and soap buff)

((Also please don't hate me for this post))

**Comment by XDTM** [comment]
That said, trivializing a job makes it boring just as much. Some additional and/or different gameplay might be more interesting. 
Mop trophies was kind of a joke but mop modifications (crafted or researched, adv mops would have more slots) might be a fancy idea to customize your janitor experience. Stuff like mopping speed, multi-tile mopping, mop-spray, chem modules, a bayonet.
Another idea might be customizing the cleanup method of some dirt types; maybe normal dirt is cleanable with water/soap, and harder stains require harsher chemicals, from space cleaner to acid. Good janitors would have a pocketful of chems to drop in the mop bucket when they need to purge the unclean. You could also do stuff like adding holy water to clean cult runes, or teslium to make tiles glow in the dark, and so on.

**Comment by Arianya** [comment]
Whatever you do, don't put buffs to cleaning on the soap. Soap is relatively easy to access and would make random people with a soap bar better at cleaning then the specialized janitor, not to mention it encourages using an item that does not make floors slippy, requires pulling and re-wetting and is easily stolen (though also relatively easy to replace, with departmental protolathes.

Any "QoL" (buffs, really) should go foremost to the mop.

**Comment by Time-Green** [comment]
i want a fucking scope and bayonet for my mop

**Comment by Kagus** [comment]
Angled grip and skeleton stock, made from real anglers and skeletons.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents.dm [hunk 4]
```diff
@@ -29,6 +29,7 @@
 	var/addiction_stage = 0
 	var/overdosed = 0 // You fucked up and this is now triggering its overdose effects, purge that shit quick.
 	var/self_consuming = FALSE
+	var/density = 1 //weight, or how far it travels when sprayed
```
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents.dm]: isn't density a byond var anyways

[the remark this answers is not in the document]
Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents.dm]: no thats dense

**Comment by Qustinnus** [comment]
can u merge https://github.com/tgstation/tgstation/pull/38310 into this pr

File: code/modules/reagents/chemistry/reagents.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kagus) [reply code/modules/reagents/chemistry/reagents.dm]: `var/thiccness = 1`

**Comment by Cyberboss** [comment]
Your PR is getting old. Either Travis CI is failing, it is still labelled work in progress (change this by adding `[READY]` to the start of your title), or there are unaddressed change requests. This may be closed soon to free up space on our tracker if it remains idle. In the event that it is, please ask in #coderbus on Rizon for it to be reopened once you have resumed work on it.

**Comment by cacogen** [comment]
I'll probably have a look at this tonight and try to take on board all of this feedback.

File: code/game/objects/items/devices/lightreplacer.dm [hunk 5]
```diff
@@ -154,6 +154,8 @@
 	Emag()
 
 /obj/item/lightreplacer/attack_self(mob/user)
+	for(var/obj/machinery/light/target in user.loc)
+		ReplaceLight(target, user)
 	to_chat(user, status_string())
```
[the remark this answers is not in the document]
Comment (cacogen) [reply code/game/objects/items/devices/lightreplacer.dm]: It always has, you want it to tell you how many bulbs you have left if you've replaced some

**Comment by cacogen** [comment]
Okay so basically it's where I want it and I've made the requested changes. I'm hoping this can get testmerged at some point.

File: code/modules/reagents/chemistry/reagents.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents.dm]: http://www.byond.com/members/?command=reference&path=atom%2Fvar%2Fdensity

It's `density` anon...

**Pull request merged**